src/Microsoft.OData.Core/UriParser/Parsers/ODataPathParser.cs

```diff
@@ -360,12 +375,39 @@ private bool TryHandleAsKeySegment(string segmentText)
             KeySegment previousKeySegment = this.FindPreviousKeySegment();
 
             KeySegment keySegment;
+
+            bool trybindingEscapeFunction = false;
+
+            if (segmentText.Length > 0 && segmentText[segmentText.Length - 1] == ':' && previous.EdmType != null)

```

Comment:
>segmentText.Length > 0 [](start = 16, length = 22)

Should this be an assert? Didn't we validate that segmentText.Length > 0 in ExtractSegmentIdentifierAndParenthesisExpression on line 118? Do we not call ExtractSegmentIdentifierAndParenthesisExpression before calling TryHanbdleAsKeySegment?

Reply:
This is the remains from a previous iteration. In my previous logic, I had permitted empty segment text. This is not needed now. 

---
In reply to: [366148775](https://github.com/OData/odata.net/pull/1621#discussion_r366148775) [](ancestors = 366148775)